builtin/credential/approle/path_role.go

```diff
@@ -994,56 +952,43 @@ func (b *backend) pathRoleCreateUpdate(ctx context.Context, req *logical.Request
 		role.SecretIDTTL = time.Second * time.Duration(data.Get("secret_id_ttl").(int))
 	}
 
-	if tokenNumUsesRaw, ok := data.GetOk("token_num_uses"); ok {
-		role.TokenNumUses = tokenNumUsesRaw.(int)
-	} else if req.Operation == logical.CreateOperation {
-		role.TokenNumUses = data.Get("token_num_uses").(int)
-	}
-	if role.TokenNumUses < 0 {
-		return logical.ErrorResponse("token_num_uses cannot be negative"), nil
-	}
-
-	if tokenTTLRaw, ok := data.GetOk("token_ttl"); ok {
-		role.TokenTTL = time.Second * time.Duration(tokenTTLRaw.(int))
-	} else if req.Operation == logical.CreateOperation {
-		role.TokenTTL = time.Second * time.Duration(data.Get("token_ttl").(int))
-	}
-
-	if tokenMaxTTLRaw, ok := data.GetOk("token_max_ttl"); ok {
-		role.TokenMaxTTL = time.Second * time.Duration(tokenMaxTTLRaw.(int))
-	} else if req.Operation == logical.CreateOperation {
-		role.TokenMaxTTL = time.Second * time.Duration(data.Get("token_max_ttl").(int))
-	}
-
-	tokenType := role.TokenType
-	if tokenTypeRaw, ok := data.GetOk("token_type"); ok {
-		tokenType = tokenTypeRaw.(string)
-		switch tokenType {
-		case "":
-			tokenType = "default"
-		case "service", "batch", "default":
-		default:
-			return logical.ErrorResponse(fmt.Sprintf("invalid 'token_type' value %q", tokenType)), nil
+	// handle upgrade cases
+	{
+		policiesRaw, ok := data.GetOk("token_policies")
+		if !ok {
+			policiesRaw, ok = data.GetOk("policies")
+			if ok {
+				role.Policies = policyutil.ParsePolicies(policiesRaw)
```

Comment:
Should we be setting `role.Policies` at all? I was wondering if we land all the values to just `role.TokenPolicies` directly.

Reply:
Same reason.